conans/model/options.py

```diff
@@ -162,19 +165,20 @@ def __init__(self, values=None):
 
         # convert tuple "Pkg:option=value", "..." to list of tuples(name, value)
         if isinstance(values, tuple):
-            new_values = []
-            for v in values:
-                option, value = v.split("=", 1)
-                new_values.append((option.strip(), value.strip()))
-            values = new_values
+            values = [item.split("=", 1) for item in values]
+
+        # convert dict {"Pkg:option": "value", "..": "..", ...} to list of tuples (name, value)
+        if isinstance(values, dict):
+            values = [(k, v) for k, v in values.items()]
 
         # handle list of tuples (name, value)
         for (k, v) in values:
+            k = k.strip()
+            v = v.strip() if isinstance(v, six.string_types) else v
```

Comment:
I still think that this line is dangerous. It will slip non-string values into the options model, which so far, was restricted. I think that it should be enforced above (in the ``dict`` input) that the values are already strings.

Reply:
That's another issue, previously non-string values could enter through the tuple interface... if we want all to be strings then something else could change (see #3620 and the ``None``)